common-caching.js

```diff
@@ -2,90 +2,145 @@
+        if (responseIsCachable(response)) {
+            event.waitUntil(cache.put(request, response.clone()));
+        }
+    }
+
+    response = replaceStrippedQsOnRedirectResponse(response, strippedParams);
+
+    return response;
 }
 
+/*
+ * Cacheability Utilities
+ */
 function requestIsCachable(request) {
-  if (hasPrivateCookie(request)) {
-    return false;
-  }
-  return true;
+    /*
+     * Given a Request, determine if it should be cached.
+     * Currently the only factor here is whether a private cookie is present.
+     */
+    return !hasPrivateCookie(request)
+}
+
+function responseIsCachable(response) {
+    /*
+     * Given a Response, determine if it should be cached.
+     * Currently the only factor here is whether the status code is cachable.
+     */
+    return CACHABLE_HTTP_STATUS_CODES.includes(response.status)
+}
+
+
+/*
+ * Request Utilities
+ */
+function stripIgnoredQuerystring(request) {
+    /**
+     * Given a Request, return a new Request with the ignored querystring keys stripped out,
+     * along with an object representing the stripped values.
+     */
+    const url = new URL(request.url);
+    const stripKeys = STRIP_QUERYSTRING_KEYS.filter((v) =>
+        url.searchParams.has(v)
+    );
+
+    let strippedParams = {};
+
+    if (stripKeys.length) {
+        stripKeys.reduce((acc, key) => {
+            acc[key] = url.searchParams.getAll(key)
+            url.searchParams.delete(key)
+            return acc
+        }, strippedParams);
+
+        return [new Request(url, request), strippedParams];
+    }
+    return [request, strippedParams];
 }
 
 function hasPrivateCookie(request) {
-  const cookieHeader = request.headers.get("Cookie");
-  if (!cookieHeader) {
-    return false;
-  }
-
-  const requestCookieNames = cookieHeader
-    .split(";")
-    .map((cookie) => cookie.split("=")[0].trim());
-
-  // Check if the request includes any of the specified 'private' cookies
-  return PRIVATE_COOKIES.some((privateCookieName) =>
-    requestCookieNames.includes(privateCookieName)
-  );
+    /*
+     * Given a Request, determine if one of the 'private' cookies are present.
+     */
+    const cookieHeader = request.headers.get("Cookie");
+    if (!cookieHeader) {
+        return false;
+    }
+
+    const requestCookieNames = cookieHeader
+        .split(";")
+        .map((cookie) => cookie.split("=")[0].trim());
+
+    return PRIVATE_COOKIES.some((privateCookieName) =>
+        requestCookieNames.includes(privateCookieName)
+    );
 }
 
-function responseIsCachable(response) {
-  if (!CACHABLE_HTTP_STATUS_CODES.includes(response.status)) {
-    return false;
-  }
-  return true;
+/**
+ * Response Utilities
+ */
+
+function replaceStrippedQsOnRedirectResponse(response, strippedParams) {
+    /**
+     * Given an existing Response, and an object of stripped querystring keys,
+     * determine if the response is a redirect.
+     * If it is, add the stripped querystrings to the location header.
```

Comment:
Should this be something opt-in? I could see this causing strange behaviour. Alternatively, should this just be something down to the downstream view rather than explicitly (and always) done here?